core/repository/sail/src/test/java/org/eclipse/rdf4j/repository/sail/SailRepositoryConnectionTest.java

```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.repository.sail;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+
+import org.eclipse.rdf4j.common.iteration.EmptyIteration;
+import org.eclipse.rdf4j.query.BooleanQuery;
+import org.eclipse.rdf4j.query.GraphQuery;
+import org.eclipse.rdf4j.query.Query;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.algebra.TupleExpr;
+import org.eclipse.rdf4j.sail.SailConnection;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link SailRepositoryConnection}
+ * 
+ * @author Jeen Broekstra
+ *
+ */
+public class SailRepositoryConnectionTest {
+
+	private SailRepositoryConnection subject;
+	private SailConnection sailConnection;
+	private SailRepository sailRepository;
+
+	@Before
+	public void setUp() throws Exception {
+		sailConnection = mock(SailConnection.class);
+		sailRepository = mock(SailRepository.class);
+
+		subject = new SailRepositoryConnection(sailRepository, sailConnection);
+	}
+
+	@Test
+	public void testPrepareQuery_not_bypassed() throws Exception {
```

Comment:
Fwiw I still haven't come up with a decent convention for this kind of test naming. Using CamelCase just quickly becomes unreadable.